libsolidity/interface/CompilerStack.cpp

```diff
@@ -841,3 +842,81 @@ string CompilerStack::computeSourceMapping(eth::AssemblyItems const& _items) con
 	}
 	return ret;
 }
+
+namespace {
+
+Json::Value gasToJson(GasEstimator::GasConsumption const& _gas)
+{
+    if (_gas.isInfinite || _gas.value > std::numeric_limits<Json::LargestUInt>::max())
+	return Json::Value(Json::nullValue);
```

Comment:
The problem I have is that `null` may or may not be serialised in the JSON hence we lose the information that a given function exists.

I am leaning towards using strings, that way both infinity and >53bits can be represented (though this is less of a real issue).

The jsonCompiler will use `null` for backwards compatibility, but the new JSON IO will have it as a string.